src/udev/udev-builtin-blkid.c

```diff
@@ -174,13 +176,17 @@ static int find_gpt_root(sd_device *dev, blkid_probe pr, bool test) {
                         if (flags & GPT_FLAG_NO_AUTO)
                                 continue;
 
-                        /* We found a suitable root partition, let's
-                         * remember the first one. */
+                        /* We found a suitable root partition, let's remember the first one, or the one with
+                         * the newest version, as determined by comparing the partition labels. */
+
+                        if (!root_id || strverscmp_improved(label, root_label) > 0) {
+                                r = free_and_strdup(&root_id, sid);
+                                if (r < 0)
+                                        return r;
 
-                        if (!root_id) {
-                                root_id = strdup(sid);
-                                if (!root_id)
-                                        return -ENOMEM;
+                                r = free_and_strdup(&root_label, label);
```

Comment:
The `root_label` is assigned, but never used, no??

Reply:
it's used in the next run around the for loop - ie, you have two root partitions, first one is found and root_label is set, when the next one is found its label is compared to the previous one (root_label) and if the version is greater then it takes over

Reply:
It's passed to strverscmp_improved() on each iteration

Reply:
Ah, I see. Thank you. I need more coffee...